pages/create/index.js

```diff
@@ -0,0 +1,94 @@
+import { useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import pets from "@/lib/pet";
+import styled from "styled-components";
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+export default function Create({ onAddPet }) {
+  const router = useRouter();
+  const [formData, setFormData] = useState({
+    name: "",
+    type: "",
+    image: "/assets/images/pets/book.png",
+  });
+
+  const [imagePath, setImagePath] = useState(formData.image);
+
+  const handleTypeChange = (event) => {
+    const selectedType = event.target.value;
+    const imagePath = `/assets/images/pets/${selectedType}.png`;
+    setFormData({ ...formData, type: selectedType, image: imagePath });
+    setImagePath(imagePath); // filled with local one!
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const newFormData = {
+      ...formData,
+      id: uuidv4(),
```

Comment:
Consider adding the `id` to each pet in your the `handleAddPet` function inside the `_app.js` file before adding the pet to the global state. 

**But WHY?** 🤨
`handleSubmit` manages the **form submission**, including validation and data preparation, while `handleAddPet` is responsible for **adding the new pet to the state**. By generating the  id with `uuid` within `handleAddPet`, you adhere to this separation, as ID generation is more aligned with state management than form processing - this concept is called **Separation of Concerns:**  

Example:
```js
import { v4 as uuidv4 } from "uuid";

// ...

function handleAddPet(newPet) {
  const petWithId = { ...newPet, id: uuidv4() };
  setMyPets([...myPets, petWithId]);
}
```